job-server/src/main/scala/spark/jobserver/AkkaClusterSupervisorActor.scala

```diff
@@ -199,16 +199,16 @@ class AkkaClusterSupervisorActor(daoActor: ActorRef) extends InstrumentedActor {
 
     logger.info("Starting context with actor name {}", contextActorName)
 
-    val contextDir: java.io.File = try {
-        createContextDir(name, contextConfig, isAdHoc, contextActorName)
-      } catch {
-        case e: Exception =>
-          failureFunc(e)
-          return
-      }
+    val driverMode = Option(config.getString("spark.jobserver.driver-mode")).getOrElse("client")
+    val (workDir, contextContent) = generateContext(name, contextConfig, isAdHoc, contextActorName)
+    logger.info("Ready to create working directory {} for context {}", workDir: Any, name)
 
     //extract spark.proxy.user from contextConfig, if available and pass it to $managerStartCommand
-    var cmdString = s"$managerStartCommand $contextDir ${selfAddress.toString}"
+    var cmdString = if (driverMode == "mesos-cluster") {
+      s"$managerStartCommand $driverMode $workDir '$contextContent' ${selfAddress.toString}"
```

Comment:
@LeightonWong Could you please explain a little that why we need single quotes around $conextContent here but not in line 210 ?

Reply:
In mesos cluster mode, jobserver manager will start in remote mode with MesosClusterDispatcher differ from local mode, they are using different submit client, I found it will cause error when parsing the contextContent which include double quotes it self, so I just add the single quotes and it works, I don't know the exact reason, may be you have to read spark MesosClusterDispatcher/MesosRestServer/RestSubmitionClient code in spark ,it takes the responsibility to start the jobserver manager int cluster mode.